benchmark/bandwagon_formula.py

```diff
@@ -0,0 +1,47 @@
+"""
+Bandwagon alpha formula.
+
+Idea:
+    We believe the 'wind' on the market.
+    The wind blows when the instrument rises/falls for consecutive x days,
+    and accumulated change rate is more than y.
+    We follow wind of instruments and positions are proportional to their
+    accumulated change rate (in the window).
+"""
+
+from thousandaire.alpha import BaseAlphaFormula
+from thousandaire.data_classes import Portfolio
+
+def catch_wind(data):
+    """
+    Do you feel the wind?
+    """
+    return (all([data[x].buy < data[x + 1].buy
+                 for x in range(-len(data), -1, 1)]) or
+            all([data[x].buy > data[x + 1].buy
+                 for x in range(-len(data), -1, 1)]))
+
+class BandWagonFormula(BaseAlphaFormula):
+    """
+    Formula for bandwagon.
+    """
+    def __init__(self, _startdate, data, parameters):
+        BaseAlphaFormula.__init__(self, _startdate, data, parameters)
+        self.rate = parameters['rate']
+        self.window = parameters['window']
+
+    def generate(self, _date, dataset):
```

Comment:
May I ask where is this "generate" method called? I checked simulator.py and simulation.py but couldn't find it :D

Reply:
It's in class BaseAlphaFormula in alpha.py. We wanted to make the formula as a function object , so we did this.
We will ask users to inherit BaseAlphaFormula while writing alpha.

Reply:
Another reason we want to seperate the callee into two steps (`__call__` and `generate`) is that we can do some pre-processing before calling users' generation function (`generate`).

Reply:
thanks 👍